src/main/java/org/ehrbase/fhirbridge/ehr/converter/specific/knownexposure/SarsCov2ExpositionEvaluationConverter.java

```diff
@@ -0,0 +1,90 @@
+package org.ehrbase.fhirbridge.ehr.converter.specific.knownexposure;
+
+import ca.uhn.fhir.rest.server.exceptions.UnprocessableEntityException;
+import org.ehrbase.fhirbridge.ehr.converter.generic.ObservationToEvaluationConverter;
+import org.ehrbase.fhirbridge.ehr.converter.generic.TimeConverter;
+import org.ehrbase.fhirbridge.ehr.opt.sarscov2expositioncomposition.definition.ExpositionVorhandenDefiningCode;
+import org.ehrbase.fhirbridge.ehr.opt.sarscov2expositioncomposition.definition.SarsCov2ExpositionEvaluation;
+import org.hl7.fhir.r4.model.Observation;
+
+public class SarsCov2ExpositionEvaluationConverter  extends ObservationToEvaluationConverter<SarsCov2ExpositionEvaluation> {
+
+    @Override
+    protected SarsCov2ExpositionEvaluation convertInternal(Observation fhirObserv) {
+        SarsCov2ExpositionEvaluation eval = new SarsCov2ExpositionEvaluation();
+
+        mapInfektionserreger(eval);
+        mapExpositionVorhanden(eval, fhirObserv);
+        mapSpezifischeDetailsZurExposition(eval, fhirObserv);
+
+        return eval;
+    }
+
+    private void mapSpezifischeDetailsZurExposition(SarsCov2ExpositionEvaluation eval, Observation fhirObserv) {
+        eval.setBeschreibungDerExpositionValue("Kontakt zu COVID-19 erkrankter Person");
+        eval.setDatumUhrzeitDerExpositionValue(TimeConverter.convertObservationTime(fhirObserv));
+    }
+
+    private void mapInfektionserreger(SarsCov2ExpositionEvaluation eval) {
+        eval.setInfektionserregerValue("SARS-CoV-2");
+    }
+
+    private void mapExpositionVorhanden(SarsCov2ExpositionEvaluation eval, Observation fhirObserv) {
+
+        Boolean codeForExpositionExists = isCodeForKnownExpositionExistent(fhirObserv);
+
+        if (Boolean.TRUE.equals(codeForExpositionExists)) {
+            mapExpositionCode(eval, fhirObserv);
+        } else {
+            mapDataAbsentReasonAsUnknownExposition(eval, fhirObserv);
+        }
+
+    }
+
+    private Boolean isCodeForKnownExpositionExistent(Observation fhirObserv) {
+
+        int numElements = fhirObserv.getValueCodeableConcept().getCoding().size();
+        switch (numElements) {
+            case 0:
+                return false;
+            case 1:
+                return true;
+            default:
+                throw new UnprocessableEntityException("Multiple known expositions reasons are not possible in SarsCov2ExpositionConverter!");
+        }
+    }
+
+    private void mapDataAbsentReasonAsUnknownExposition(SarsCov2ExpositionEvaluation eval, Observation fhirObserv) {
+        int numElements = fhirObserv.getDataAbsentReason().getCoding().size();
+
+        if (1 != numElements) {
+            throw new UnprocessableEntityException("Zero or Multiple data absent reasons are not possible in SarsCov2ExpositionConverter with no valid exposition information!");
```

Comment:
its better to check if this element is existent at all, if not nullpointer will be thrown.
if (fhirObserv.hasValueCodeableConcept && ¨¨.hasCoding)
int num Elements = .. ;
make numElements an Optional and check then if its empty if yes throw the error.

Reply:
Ist geändert, ich bitte dich um Kontrolle.